lib/exercism/team.rb

```diff
@@ -56,45 +55,50 @@ def defined_with(options, inviter=nil)
     tags = [options[:slug], options[:name], options[:tags]].join(',')
     self.tags = Tag.create_from_text(tags)
 
-    recruits = User.find_or_create_in_usernames(potential_recruits(options[:usernames])) if options[:usernames]
-    recruits = options[:users] if options[:users]
-
-    if recruits
-      recruits.each do |recruit|
-        TeamMembership.create(user: recruit, team: self, inviter: inviter)
-      end
-    end
+    users = User.find_or_create_in_usernames(potential_members(options[:usernames])) if options[:usernames]
+    users = options[:users] if options[:users]
+    invite(users, inviter)
 
     self
   end
 
-  def recruit(usernames, inviter)
-    recruits = User.find_or_create_in_usernames(potential_recruits(usernames.to_s)) - all_members
+  def invite_with_usernames(usernames, inviter)
+    invite(
+      User.find_or_create_in_usernames(potential_members(usernames)),
+      inviter
+    )
+  end
+
+  def invite(users, inviter)
+    return unless users.present?
+
+    users = Array(users) - all_members
 
-    recruits.each do |recruit|
-      TeamMembership.create(user: recruit, team: self, inviter: inviter)
+    users.each do |user|
+      TeamMembershipInvite.create(user: user, team: self, inviter: inviter, refused: false)
     end
   end
 
+  def accept_invite(username)
```

Comment:
I know the feeling :)
I suggest trying it and seeing what happens, you can always revert.

One thing that I noticed was that the "team centric" thinking that begins in the route sends you in the wrong direction.

Instead of:
```
team.accept_invite(current_user.username)
```

How about:
```
invite_for_team_and_user( slug, current_user.id ).accept
```